src/main/java/edu/cursor/spring_hw1/conroller/AuthorController.java

```diff
@@ -0,0 +1,54 @@
+package edu.cursor.spring_hw1.conroller;
+
+import edu.cursor.spring_hw1.entities.Author;
+import edu.cursor.spring_hw1.entities.Book;
+import edu.cursor.spring_hw1.exceptions.AuthorNotFoundException;
+import edu.cursor.spring_hw1.repository.AuthorRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("library/author")
+public class AuthorController {
+    private final AuthorRepository authorRepository;
+
+    @Autowired
+    public AuthorController(AuthorRepository authorRepository) {
+        this.authorRepository = authorRepository;
+    }
+
+    @GetMapping
+    public String getAuthors() {
+        return authorRepository.getAuthors().toString();
+    }
+
+    @GetMapping("/booksOf/{id}")
+    public List<Book> getAuthorBooks(@PathVariable Integer id){
+        return authorRepository.getAuthors().get(id).getBooks();
+    }
+
+    @GetMapping("{id}")
+    public Author getAuthor(@PathVariable Integer id) {
+        if (id.compareTo(authorRepository.getAuthors().size()) >= 0) {
+            throw new AuthorNotFoundException("author with id " + id + " not found");
+        }
+        return authorRepository.getAuthors().get(id);
+    }
+
+    @PostMapping
+    public void createAuthor(@RequestBody Author author) {
+        authorRepository.addAuthor(author);
+    }
+
+    @DeleteMapping("{id}")
+    public void deleteAuthor(@PathVariable Integer id) {
+        authorRepository.deleteAuthor(id);
+    }
+
+    @PutMapping("{id}")
```

Comment:
POST `library/authors/{id}`
`@RequestBody Book book`

Reply:
there`s nothing to fix